File: Data Structures & Algorithms/valid-binary-search-tree/submission-4.js
class Solution {
	/**
	 * @param {TreeNode} root
	 * @return {boolean}
	 */
	isValidBST(root) {
		function validate(node, mn, mx) {
			if (!node) return true;
			if (node.val <= mn)
				return false;
			if (node.val >= mx)
				return false;
			return validate(node.right, node.val, mx) &&
				validate(node.left, mn, node.val);
		}
		return validate(root, -Infinity, Infinity);

	}
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-4.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> ans;
        return helper(root ,&ans );

        
    }
    vector<int> helper(TreeNode* curNode , vector<int>*ans){
        if(curNdoe == nullptr){
            return ans;
        }
        ans.push_back(curNode->right);
        helper(curNode->right , ans);
    }
};
